lib/modules/datasource/unity3d/index.ts

```diff
@@ -0,0 +1,73 @@
+import { XmlDocument } from 'xmldoc';
+import { cache } from '../../../util/cache/package/decorator';
+import * as Unity3dVersioning from '../../versioning/unity3d';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+export class Unity3dDatasource extends Datasource {
+  static readonly streams: { [key: string]: string } = {
+    lts: 'https://unity.com/releases/editor/lts-releases.xml',
+    stable: 'https://unity.com/releases/editor/releases.xml',
+    beta: 'https://unity.com/releases/editor/beta/latest.xml',
+  };
+
+  static readonly id = 'unity3d';
+
+  override readonly defaultRegistryUrls = [
+    Unity3dDatasource.streams.stable,
+    Unity3dDatasource.streams.lts,
+  ];
+
+  override readonly defaultVersioning = Unity3dVersioning.id;
+
+  override readonly registryStrategy = 'merge';
+
+  constructor() {
+    super(Unity3dDatasource.id);
+  }
+
+  getByStream = async (
+    registryUrl: string | undefined,
+    withHash: boolean,
+  ): Promise<ReleaseResult | null> => {
+    const response = await this.http.get(registryUrl!);
+    const document = new XmlDocument(response.body);
+    const releases = document
+      .childNamed('channel')!
+      .childrenNamed('item')
+      .map((itemNode) => {
+        const versionWithHash = `${itemNode.childNamed('title')?.val} (${itemNode.childNamed('guid')?.val})`;
+        const versionWithoutHash = itemNode.childNamed('title')?.val;
```

Comment:
Handle missing channel objects and missing titles.

Reply:
Added tests in 4b48d711bd6f1032be72267ba4c45dad6759e4a8 and dd49d63a408d2792c46e9ab0ffc6f61dbc68c471.